.github/workflows/build_and_test.yml

```
@@ -151,7 +163,66 @@ jobs:
             PACKAGE=${pkg}${{ env.VERSION }}.zip
             curl -i -u "${{ secrets.connect-auth-user }}:${{ secrets.connect-auth-token }}" -F package=@$PACKAGE "$CONNECT_URL/site/marketplace/upload?batch=true"
           done
-  
+
+      # --- Prepare Docker build context ---
+      - name: Prepare Docker build context
+        if: ${{ inputs.preview }}
+        run: |
+          echo "=== Copying available package zips to docker directory ==="
+          # Ensure the nuxeo-ai-core package is always copied (using dynamic version)
+          if [ -f "nuxeo-ai-core-package/target/nuxeo-ai-core-${{ env.VERSION }}.zip" ]; then
+            echo "Copying nuxeo-ai-core package..."
+            cp "nuxeo-ai-core-package/target/nuxeo-ai-core-${{ env.VERSION }}.zip" docker/
+          else
+            echo "ERROR: nuxeo-ai-core package not found at nuxeo-ai-core-package/target/nuxeo-ai-core-${{ env.VERSION }}.zip"
+            exit 1
+          fi
+          
+          # Copy other available packages (optional) - use dynamic naming
+          find addons/nuxeo-ai-image-quality-package/target/ -name "*package*.zip" -exec cp {} docker/ \; 2>/dev/null || echo "image-quality package not found"  
+          find addons/nuxeo-ai-aws-package/target/ -name "*package*.zip" -exec cp {} docker/ \; 2>/dev/null || echo "aws package not found"
+          find addons/nuxeo-ai-gcp-package/target/ -name "*package*.zip" -exec cp {} docker/ \; 2>/dev/null || echo "gcp package not found"
+          
+          # Copy any other built packages dynamically
+          find . -path "*/target/*package*.zip" -not -path "./docker/*" -exec cp {} docker/ \; 2>/dev/null || echo "No additional packages found"
```

This will only fail if `find`'s search directory does not exist. If it exists but no package zip is present it won't fail, and hence won't reach `echo` 

```suggestion
          find . -path "*/target/*package*.zip" -exec cp {} docker/ \;
          echo "List of packages embedded in preview image:"
          ls -l docker/*package*.zip
```